include/nbl/builtin/hlsl/colorspace/EOTF.hlsl

```diff
@@ -0,0 +1,114 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+
+#include <nbl/builtin/hlsl/common.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace colorspace
+{
+namespace eotf
+{
+
+float3 identity(in float3 nonlinear)
+{
+    return nonlinear;
+}
+
+float3 impl_shared_2_4(in float3 nonlinear, in float vertex)
+{
+    bool3 right = (nonlinear > vertex.xxx);
+    return lerp(nonlinear / 12.92, pow((nonlinear + 0.055.xxx) / 1.055, (2.4).xxx), right);
+}
+
+// compatible with scRGB as well
+float3 sRGB(in float3 nonlinear)
+{
+    bool3 negatif = (nonlinear < (0.0).xxx);
+    float3 absVal = impl_shared_2_4(abs(nonlinear), 0.04045);
+    return lerp(absVal, -absVal, negatif);
+}
+
+// also known as P3-D65
+float3 Display_P3(in float3 nonlinear)
+{
+    return impl_shared_2_4(nonlinear, 0.039000312);
+}
+
+
+float3 DCI_P3_XYZ(in float3 nonlinear)
+{
+    return pow(nonlinear * 52.37, (2.6).xxx);
+}
```

Comment:
1. remove `in` and replace with `const`
2. triple check that all functions are functionally identical to the GLSL and that the GLSL hasn't been updated since
3. refactor all free functions into functor structs with `static operator()`

Reply:
(3) is needed so I can do
```cpp
template<class OETF>
struct ACES
{
};
```